cmd/snap/cmd_snap_op.go

```diff
@@ -353,6 +355,9 @@ func (x *cmdInstall) installOne(name string, opts *client.SnapOptions) error {
 		installFromFile = true
 		changeID, err = cli.InstallPath(name, opts)
 	} else {
+		if opts.Instance != "" {
```

Comment:
Thinking about this, why not? I mean: `snap install go_stable` and `snap install go --instance=stable` would simply be the same if we allow this, no? (I don't have a strong opinion either way it just seems slightly odd to forbid it - unless I miss something of course).

Reply:
Whether to include the snap name in `--instance` is also a good question. I'm a bit on the fence about including snap name there. On one hand `snap install go --instance stable` reads fine, but then the snap is named `go_stable` (unexpectedly?). OTOH `snap install go --instance go_stable` matches the actual snap name used by snapd, making it more explicit? Also, this will match the plain `snap install go_stable` command. 

BTW. we could use `snap install go --name go_stable` then it's explicit and `go_stable` as snap name is not unexpected.